gradle/spotbugs.gradle

```diff
@@ -2,11 +2,35 @@ allprojects {
   apply plugin: 'com.github.spotbugs'
 
   spotbugs {
-    ignoreFailures = true
+    ignoreFailures = false
     reportLevel = "high"
     omitVisitors = ["FindDeadLocalStores"]
     effort = "max"
     excludeFilter = file("$rootDir/gradle/spotbugs-exclude.xml")
+    toolVersion = '4.1.3'
+  }
+
+  if (!isCI) {
+    spotbugsTest {
```

Comment:
What of these are default configuration anyway and what specifically you want to override?

Reply:
Default config is to generate an XML file for machine consumption, makes sense on a CI environment, not as much sense on a local machine.

Reply:
Then I believe this whole `if` block can be replaced with 
```
  spotbugsTest {
    reports {
      html.enabled = !isCI
    }
  }
```